superduperdb/backends/mongodb/query.py

```diff
@@ -798,3 +811,42 @@ def model_update(
             collection_name = f'_outputs.{key}.{model}'
             collection = db.databackend.get_table_or_collection(collection_name)
             collection.bulk_write(bulk_docs)
+
+
+def _get_decode_function(db) -> t.Callable[[t.Any], t.Any]:
+    def decode(output):
+        schema_identifier = output.get(SCHEMA_KEY)
+        if schema_identifier is None:
+            return output
+        schema = get_schema(db, schema_identifier)
+        for k in output.keys():
+            if field := schema.fields.get(k):
+                output[k] = field.decode_data(output[k])
+        return output
+
+    return decode
+
+
+_SCHEMA_CACHE = {}
```

Comment:
We have the caches in the `Datalayer` - we can add it there.

Reply:
Great, I will add it here

Reply:
Done